src/update/index.js

```diff
@@ -56,8 +77,8 @@ class UpdateController {
 
   async update () {
     if (!this.shouldUpdate) {
-      const { shouldUpdate } = await this.getLatestVersion()
-      if (!shouldUpdate) return
+      await this.check()
```

Comment:
should this be const shouldUpdate = await this.check() ?

seems like checking this.shouldUpdate is checking a byproduct of the check method instead of just the returned value

Reply:
you right